source/dotnet/Library/AdaptiveCards.Rendering.Wpf/AdaptiveRenderContext.cs

```diff
@@ -195,7 +195,21 @@ public FrameworkElement Render(AdaptiveTypedElement element)
                 return rendered;
             }
 
+            // Since no renderer exists for this element, add warning and render fallback (if available)
             Warnings.Add(new AdaptiveWarning(-1, $"No renderer for element '{element.Type}'"));
```

Comment:
what happens to these Warnings? Do they end up on cosmos or something ?

Reply:
warnings are part of the output from rendering an adaptive card (see `ParseResult`). it's up to the renderer to handle them as they see fit. for our visualizers, we show the warnings alongside the rendered card to aid card authors. I'm not sure, but I'd guess that some consumers (e.g. Teams) feed the warnings into their own telemetry channels.

---
In reply to: [272391324](https://github.com/Microsoft/AdaptiveCards/pull/2560#discussion_r272391324) [](ancestors = 272391324)

Reply:
i see thanks

---
In reply to: [272392569](https://github.com/Microsoft/AdaptiveCards/pull/2560#discussion_r272392569) [](ancestors = 272392569,272391324)